src/components/VmDetails/cards/SnapshotsCard/NewSnapshotModal.js

```diff
@@ -78,46 +80,59 @@ class NewSnapshotModal extends Component {
           { msg.createSnapshot() }
         </a>
 
-        <Modal show={this.state.showModal} onHide={this.close} dialogClassName={style['create-snapshot-container']} id={modalId}>
-          <Modal.Header>
-            <button
-              className='close'
+        <Modal
+          isOpen={this.state.showModal}
+          onClose={this.close}
+          id={modalId}
+          title={ msg.createSnapshot() }
+          variant={ModalVariant.small}
+          position='top'
+          actions={[
+            <Button key='save' variant='primary' onClick={this.handleSave} id={`${modalId}-create`}>
+              { msg.create() }
+            </Button>,
+            <Button
+              variant='link'
               onClick={this.close}
-              aria-hidden='true'
-              aria-label={msg.close()}
-              id={`${modalId}-close`}
+              id={`${modalId}-cancel`}
+              key='cancel'
             >
-              <Icon type='pf' name='close' />
-            </button>
-            <Modal.Title>{ msg.createSnapshot() }</Modal.Title>
-          </Modal.Header>
-          <Modal.Body>
-            <Form onSubmit={this.handleSave} horizontal>
-              <Col sm={12}>
-                <Alert type='info'>
-                  { msg.snapshotInfo() }
-                </Alert>
-              </Col>
-              <FormGroup bsClass='form-group col-sm-12 required' validationState={this.state.emptyDescription ? 'error' : null}>
-                <label className='col-sm-3 control-label'>
-                  { msg.name() }
-                </label>
-                <div className='col-sm-9'>
-                  <FormControl
-                    type='text'
-                    value={this.state.description}
-                    onChange={this.handleDescriptionChange}
-                    id={`${modalId}-description-edit`}
-                  />
-                  {
+              { msg.cancel() }
+            </Button>,
+
+          ]}
+        >
+          <Form onSubmit={this.handleSave} horizontal>
+            <Col sm={12}>
+              <Alert
+                variant='info'
+                isInline
+                title={msg.details()}
+                style={{ 'margin-bottom': '10px' }}
+              >
+                {msg.snapshotInfo() }
+              </Alert>
+            </Col>
+            <FormGroup bsClass='form-group col-sm-12 required' validationState={this.state.emptyDescription ? 'error' : null}>
```

Comment:
May need to add some more cols to the labels so the "Save Memory" label doesn't wrap (in English at least)

Reply:
Dialogs with rich content (i.e. forms) required more refactorings. This dialogs was fixed together in with VM Details screen (or later). In the most recent PR #1592  this looks OK.